airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DefaultConfigPersistence.java

```diff
@@ -75,26 +85,127 @@ public DefaultConfigPersistence(final Path storageRoot, final JsonSchemaValidato
   @Override
   public <T> void writeConfig(ConfigSchema configType, String configId, T config) throws JsonValidationException, IOException {
     synchronized (lock) {
-      writeConfigInternal(configType, configId, config);
+      writeConfigInternal(configType, configId, config, Optional.empty());
     }
   }
 
-  private <T> T getConfigInternal(ConfigSchema configType, String configId, Class<T> clazz)
-      throws ConfigNotFoundException, JsonValidationException, IOException {
+  private <T> void writeConfig(ConfigSchema configType, String configId, T config, Path rootOverride) throws JsonValidationException, IOException {
+    writeConfigInternal(configType, configId, config, Optional.of(rootOverride));
+  }
+
+  private <T> void writeConfigs(ConfigSchema configType, Stream<T> configs, Path rootOverride) {
+    configs.forEach(config -> {
+      String id = configType.getId(config);
+      try {
+        writeConfig(configType, id, config, rootOverride);
+      } catch (JsonValidationException | IOException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  @Override
+  public Map<String, Stream<JsonNode>> dump() throws IOException {
+    final Map<String, Stream<JsonNode>> configs = new HashMap<>();
+
+    final List<String> directories = listDirectories();
+    for (String directory : directories) {
+      final List<JsonNode> configList = listConfig(directory);
+      configs.put(directory, configList.stream());
+    }
+    return configs;
+  }
+
+  private List<String> listDirectories() throws IOException {
+    try (Stream<Path> files = Files.list(storageRootWithConfigDirectory)) {
+      return files.map(c -> c.getFileName().toString()).collect(Collectors.toList());
+    }
+  }
+
+  private List<JsonNode> listConfig(String configType) throws IOException {
+    final Path configTypePath = storageRootWithConfigDirectory.resolve(configType);
+    if (!Files.exists(configTypePath)) {
+      return Collections.emptyList();
+    }
+    try (Stream<Path> files = Files.list(configTypePath)) {
+      final List<String> ids = files
+          .filter(p -> !p.endsWith(".json"))
+          .map(p -> p.getFileName().toString().replace(".json", ""))
+          .collect(Collectors.toList());
+
+      final List<JsonNode> configs = Lists.newArrayList();
+      for (String id : ids) {
+        try {
+          final Path configPath = storageRootWithConfigDirectory.resolve(configType).resolve(String.format("%s.json", id));
+          if (!Files.exists(configPath)) {
+            throw new RuntimeException("Config NotFound");
+          }
+
+          final JsonNode config = Jsons.deserialize(Files.readString(configPath), JsonNode.class);
+          configs.add(config);
+        } catch (RuntimeException e) {
+          throw new IOException(e);
+        }
+      }
+
+      return configs;
+    }
+  }
+
+  @Override
+  public <T> void atomicConfigImport(Map<ConfigSchema, Stream<T>> configs, boolean dryRun)
```

Comment:
This interface will be clearer if you call it `importConfigs` or `replaceAllConfigs`. 

Right now `atomic` doesn't really help understand what happens in the code.